File: frontend/lib/models/tracking.dart
class TreeTracking {
  final int id;
  final int diagnosisId;
  final DateTime tanggal;
  final int tingkatKesehatan;
  final String? lokasi;
  final String? petugas;
  final String? catatan;
  final String? tindakanDilakukan;
  final String? fotoUrl;

  TreeTracking({
    required this.id,
    required this.diagnosisId,
    required this.tanggal,
    required this.tingkatKesehatan,
    this.lokasi,
    this.petugas,
    this.catatan,
    this.tindakanDilakukan,
    this.fotoUrl,
  });

  factory TreeTracking.fromJson(Map<String, dynamic> json) {
    return TreeTracking(
      id: json['id'],
      diagnosisId: json['diagnosis_id'],
      tanggal: DateTime.parse(json['tanggal']),
      tingkatKesehatan: json['tingkat_kesehatan'],
      lokasi: json['lokasi'],
      petugas: json['petugas'],
      catatan: json['catatan'],
      tindakanDilakukan: json['tindakan_dilakukan'],
      fotoUrl: json['foto_url'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'diagnosis_id': diagnosisId,
      'tanggal': tanggal.toIso8601String(),
      'tingkat_kesehatan': tingkatKesehatan,
      'lokasi': lokasi,
      'petugas': petugas,
      'catatan': catatan,
      'tindakan_dilakukan': tindakanDilakukan,
      'foto_url': fotoUrl,
    };
  }
}

File: backend/schemas/tracking.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class TrackingBase(BaseModel):
    pohon_id: int
    tingkat_kesehatan: float
    catatan: Optional[str] = None
    foto_url: Optional[str] = None
    petugas: Optional[str] = None
    lokasi: Optional[str] = None
    tindakan: Optional[str] = None

class TrackingCreate(TrackingBase):
    tanggal: Optional[datetime] = None

class TrackingResponse(TrackingBase):
    id: int
    tanggal: datetime

    class Config:
        orm_mode = True

class HistoryBase(BaseModel):
    diagnosis_id: int
    lokasi: Optional[str] = None
    petugas: Optional[str] = None
    catatan: Optional[str] = None
    tindakan_lanjutan: Optional[str] = None
    status_penanganan: Optional[str] = None
    foto_url: Optional[str] = None
    is_resolved: bool = False

class HistoryCreate(HistoryBase):
    pass

class HistoryResponse(HistoryBase):
    id: int
    waktu: datetime
    is_resolved: bool

    class Config:
        orm_mode = True